### PR TITLE
Remove deprecated code

### LINKED ISSUE #1698: Remove deprecated features
This issue is to track deprecated features that are scheduled for removal in asdf `4.0.0`:
- [ ] `asdf.util.human_list` (deprecated: #1688)
- [ ] `asdf.util.resolve_name` (deprecated: #1688)
- [ ] `asdf.util.minversion` (deprecated: #1688)
- [ ] `asdf.util.iter_subclasses` (deprecated: #1688)

### LINKED ISSUE #1859: Add `copy_arrays` removal to "what's new"
https://github.com/asdf-format/asdf/pull/1800 removed `copy_arrays` let's add it to "what's new" once https://github.com/asdf-format/asdf/pull/1852 is merged.

### PR DESCRIPTION
**Pull request by braingram**
# Description

This PR removes deprecated API. See docs link below for more details.

Closes #1698
Closes #1859

It needs:
- [x] docs (including the start of the 4.0 what's new): https://asdf--1852.org.readthedocs.build/en/1852/asdf/whats_new.html#whats-new-4-0-0
- [x] address "TODO"s mentioned below (splitting them into separate issues). I will open these when this PR is approved but prior to merging since we will likely have a discussed plan at that point.

The major changes to make sure to note are:
- no more namedtuple support (so we can handle these with Extensions/Converters)
- use of a tag uri for a schema `$ref` will no longer resolve
- `resolve_references` is no longer called for `AsdfFile.__init__` or `asdf.open`

# Checklist:

- [x] pre-commit checks ran successfully
- [x] tests ran successfully
- [x] for a public change, added a [towncrier news fragment](https://towncrier.readthedocs.io/en/stable/tutorial.html#creating-news-fragments) <details><summary>`changes/<PR#>.<changetype>.rst`</summary>

    - ``changes/<PR#>.feature.rst``: new feature
    - ``changes/<PR#>.bugfix.rst``: bug fix
    - ``changes/<PR#>.doc.rst``: documentation change
    - ``changes/<PR#>.removal.rst``: deprecation or removal of public API
    - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  </details>
- [x] for a public change, updated documentation
- [x] for any new features, unit tests were added

### CODE REVIEW

File: asdf/schema.py [hunk 1]
```diff
@@ -415,8 +406,7 @@ def load_schema(url, resolver=None, resolve_references=False):
         If ``True``, resolve all ``$ref`` references.
 
     """
-    if resolver is None:
-        resolver = _tag_to_uri
+    # TODO deprecate resolver?
```
Comment (braingram) [inline asdf/schema.py:409]: Part of me wonders if this is sufficient reason to delay 4.0 for another 3.x release.

`asdf.schema.load_schema` is a widely used part of the public API. However the `resolver` argument is I think unused (a more thorough look is needed). Keeping it significantly complicates this file. It was in place in part to allow `$ref` entries in schema files to resolve when provided `tag:foo:bar...` uris (which has long been deprecated and is being removed in asdf 4.0).

The other part of me knows that there is bound to be something we "miss" in the API cleanup and this so far is it.

File: asdf/schema.py [hunk 2]
```diff
@@ -537,6 +527,7 @@ def get_validator(
     -------
     validator : jsonschema.Validator
     """
+    # TODO deprecate url_mapping
```
Comment (braingram) [inline asdf/schema.py:530]: I think we can just remove this as this function isn't part of the documented API. I would first like to double check it's not in use by our downstream and to handle this in a separate PR. I'll open and issue and remove this TODO as this PR progresses.

Comment (braingram) [reply asdf/schema.py:530]: https://github.com/asdf-format/asdf/issues/1864

**Review by nden — APPROVED** [review]
LGTM, assuming the weldex failure is not an issue.

**Comment by braingram** [comment]
> LGTM, assuming the weldex failure is not an issue.

Thanks! There is a PR that fixes the error:
https://github.com/BAMWelDX/weldx/pull/910

I may open a smaller one that's just that error since the above PR also has schema changes (although those are required for https://github.com/asdf-format/asdf/pull/1744).

**Pull request merged**